configuration/salt.md

```diff
@@ -243,13 +354,67 @@ Template name on which given VM is based (if any).
 
 VM which provides network to the given VM
 
+## Debugging
+The output for each vm is logged in `/var/log/qubes/mgmt-VM_NAME.log`.
+
+If the log does not contain useful information, you can stop `qubesctl` by 
+pressing `ctrl+z`.
+
+You need to:
+1. run `sudo qubesctl --skip-dom0 --target=VM_NAME state.highstate`
+2. When your vm is being started (yellow) press Ctrl-Z on qubesctl.
+3. Open terminal in disp-mgmt-VM_NAME.
+4. Look at /etc/qubes-rpc/qubes.SaltLinuxVM - this is what is 
+   executed in the management vm.
+5. Get the last two lines:
+
+    export PATH="/usr/lib/qubes-vm-connector/ssh-wrapper:$PATH"
+    salt-ssh "$target_vm" $salt_command
+
+  Adjust $target_vm (VM_NAME) and $salt_command (state.highstate).
+6. Execute them, fix problems, repeat.
+
+## Known pitfalls
+### Using fedora-24-minimal
+The fedora-24-minimal package is missing the sudo package.
+You can install it via:
+
+    qvm-run -p vmname 'dnf install -y sudo'
+    
+The `-p` is will cause the execution to wait until the package is installed.
+This is important when using a state with `cmd.run`.
+
+### Disk quota exceeded (when installing templates)
+If you install multiple templates you may encounter this error.
+The solution is to shut down the updatevm between each install.
+E.g.:
+
+    install template and shutdown updatevm:
+      cmd.run:
+      - name: sudo qubes-dom0-update -y fedora-24; qvm-shutdown {{salt.cmd.run(qubes-prefs updatevm) }}
```

Comment:
> am i correct in assuming pkg.installed fromrepo is handled correctly for dom0?

Yes.

> is there a way to pass the --clean flag via salt? (or is this done automatically?(if not, i think this would be a good choice))

It's done when you specify `refresh: True`